AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java

```diff
@@ -379,4 +397,35 @@ private void handleAudioPermissionsDenied() {
     private void hidePermissionsWarning() {
         setupMissingLayout.setVisibility(View.GONE);
     }
+
+    private void initializeSpeaker() {
+        ((AzureCalling) getApplication()).createAudioSessionManager();
+        audioSessionManager = ((AzureCalling) getApplicationContext()).getAudioSessionManager();
+        // By default, turn on the speaker
+        audioSessionManager.switchAudioDeviceType(AudioDeviceType.SPEAKER);
+    }
+
+    @Override
+    public boolean dispatchTouchEvent(final MotionEvent ev) {
+        if (ev.getAction() == MotionEvent.ACTION_DOWN) {
+            final View v = getCurrentFocus();
+            if (isShouldHideInput(v, ev)) {
+                final InputMethodManager imm = (InputMethodManager) getSystemService(
+                        Context.INPUT_METHOD_SERVICE);
+                imm.hideSoftInputFromWindow(v.getWindowToken(), 0);
+            }
+        }
+        return super.dispatchTouchEvent(ev);
+    }
+
+    private boolean isShouldHideInput(final View v, final MotionEvent event) {
+        if (v != null && (v instanceof EditText)) {
+            final int[] l = {0, 0};
+            v.getLocationInWindow(l);
+            final int left = l[0], top = l[1], bottom = top + v.getHeight(), right = left + v.getWidth();
+            return !(event.getX() > left) || !(event.getX() < right)
+                    || !(event.getY() > top) || !(event.getY() < bottom);
+        }
+        return false;
+    }
```

Comment:
I think we can simply this bit of code that dismisses the keyboard with the following:
https://stackoverflow.com/a/19828165

Basically, just add onFocusChangeListener to the setupName and the audio device button:
``` java
setupName.setOnFocusChangeListener((view, hasFocus) -> {
            if (hasFocus) {
                setupEnter.setTextColor(ContextCompat.getColor(this, R.color.colorAccent));
            } else {
                setupEnter.setTextColor(ContextCompat.getColor(this, R.color.textbox_secondary));
                final InputMethodManager inputMethodManager =
                        (InputMethodManager) getSystemService(Activity.INPUT_METHOD_SERVICE);
                inputMethodManager.hideSoftInputFromWindow(view.getWindowToken(), 0);
            }
        });

deviceOptionsButton.setOnFocusChangeListener((view, hasFocus) -> {
            if (hasFocus) {
                openAudioDeviceList();
            }
        });
```

And then just remember to add the following to the parent ConstraintLayout:
``` xml
android:focusableInTouchMode="true"
android:clickable="true"
```

Reply:
Thanks for investigation of the part of logic simplification. I updated the code to address it.